scripts/nlp_language_modeling/merge_lora_weights/merge.py

```diff
@@ -211,14 +216,16 @@ def main(cfg) -> None:
     # trainer required for restoring model parallel models
     trainer = Trainer(strategy=NLPDDPStrategy(), **cfg.trainer)
 
-    if cfg.tensor_model_parallel_size < 0 or cfg.pipeline_model_parallel_size < 0:
-        model_config = MegatronGPTModel.restore_from(
-            restore_path=cfg.gpt_model_file, trainer=trainer, return_config=True,
-        )
+    model_config = MegatronGPTModel.restore_from(
+        restore_path=cfg.gpt_model_file,
+        trainer=trainer,
+        return_config=True,
+    )
 
-        with open_dict(cfg):
-            cfg.tensor_model_parallel_size = model_config.get('tensor_model_parallel_size', 1)
-            cfg.pipeline_model_parallel_size = model_config.get('pipeline_model_parallel_size', 1)
+    with open_dict(cfg):
+        # load base model with TP1 PP1.
+        model_config.tensor_model_parallel_size = 1
+        model_config.pipeline_model_parallel_size = 1
```

Comment:
Maybe add warning message for users to use CPU when base model is too large for tp1pp1?